orttraining/orttraining/python/training/ortmodule/_utils.py

```diff
@@ -25,6 +25,24 @@
 import types
 import warnings
 from distutils.version import LooseVersion
+import random
+import numpy as np
+
+def get_random_states():
+    r_state = random.getstate()
+    np_state = np.random.get_state()
+    torch_state = torch.get_rng_state()
+    torch_cuda_state = torch.cuda.get_rng_state() if torch.cuda.is_available() else None
```

Comment:
Could there be other libraries that we would need to set in the future? Is there a way to avoid having to do this? Or maybe let user control this with using `_use_deterministic_algorithms`?

Reply:
Probably `_use_deterministic_algorithms` has a different purpose than the scope of this pull request. I think it is ok to go with the assumptions you have set out so far.